service/controller/clusterapi/v27/key/cluster.go

```diff
@@ -18,6 +18,10 @@ func ClusterID(cluster v1alpha1.Cluster) string {
 	return providerStatus(cluster).Cluster.ID
 }
 
+func ClusterIsDeleted(cluster v1alpha1.Cluster) bool {
+	return cluster.GetDeletionTimestamp() != nil
+}
```

Comment:
This should work for both https://github.com/giantswarm/release-operator/blob/a51636ddc268f6936d3a61d08ace640aad4a2bb8/service/controller/key/key.go#L28-L34

Reply:
Cool. That was too obvious for me. I will rework that.